service/src/main/java/bio/terra/workspace/service/resource/controlled/flight/clone/workspace/LaunchCloneControlledFlexibleResourceFlightStep.java

```diff
@@ -0,0 +1,97 @@
+package bio.terra.workspace.service.resource.controlled.flight.clone.workspace;
+
+import static bio.terra.workspace.common.utils.FlightUtils.validateRequiredEntries;
+
+import bio.terra.stairway.FlightContext;
+import bio.terra.stairway.FlightMap;
+import bio.terra.stairway.Step;
+import bio.terra.stairway.StepResult;
+import bio.terra.stairway.StepStatus;
+import bio.terra.stairway.exception.DatabaseOperationException;
+import bio.terra.stairway.exception.DuplicateFlightIdException;
+import bio.terra.stairway.exception.RetryException;
+import bio.terra.stairway.exception.StairwayExecutionException;
+import bio.terra.workspace.service.iam.AuthenticatedUserRequest;
+import bio.terra.workspace.service.job.JobMapKeys;
+import bio.terra.workspace.service.resource.controlled.cloud.any.flexibleresource.ControlledFlexibleResource;
+import bio.terra.workspace.service.resource.controlled.flight.clone.flexibleresource.CloneControlledFlexibleResourceFlight;
+import bio.terra.workspace.service.workspace.flight.WorkspaceFlightMapKeys;
+import java.util.UUID;
+import javax.annotation.Nullable;
+
+public class LaunchCloneControlledFlexibleResourceFlightStep implements Step {
```

Comment:
So, I tested this locally with a few cases via Swagger. 

Most importantly: I had flex resource A with `COPY_RESOURCE`, and flex resource B with `COPY_NOTHING`. 

Resource A copied over, and resource B did not copy over (thus, the correct behavior).

Do we need to add tests confirming the flex resources are cloned over during the workspace clone flight?

Reply:
You already validate the resource clone operation. So the additional piece to cover is making sure that the resources are included in the clone. Perhaps add a flex resource to the integration `CloneWorkspace` test. I think the test structure is pretty obvious.